script.js

```diff
@@ -0,0 +1,364 @@
+initializeCalculator();
+function initializeCalculator() {
+  let first = {value: "", unsaved: true, saved: false, integer: true};
+  let second = {value: "", unsaved: true, saved: false, integer: true};
+  let operator = {value: "", unsaved: true, saved: false};
+  let result = "";
+  let allowNewNumber = false;
+
+  listenClicks();
+  listenKeys();
+
+  function listenKeys() {
+    window.addEventListener("keydown", event => {
+      const key = document.querySelector(`[data-key="${event.key}"]`);
+      if (key) {
+        bindEvent(key.id);
+        key.focus(); // Safari and Firefox don't handle focus in CSS files
+      }
+    });
+  }
+
+  function listenClicks() {
+    const buttons = document.querySelectorAll(".btn");
+    buttons.forEach(button => {
+      button.addEventListener("click", event => {
+        const click = event.target;
+        bindEvent(click.id);
+        click.focus(); // Safari and Firefox don't handle focus in CSS files
+      });
+    });
+  }
+
+  function bindEvent(event) {
+    if (["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].includes(event)) {
```

Comment:
To make it more readable, I'd add `["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]` as an array and call it something like `digits`. Same with `["+", "-", "÷", "×", "^"]` and call it operands.